backend/kesaseteli/applications/api/v1/views.py

```diff
@@ -145,7 +147,10 @@ def retrieve(self, request, *args, **kwargs):
         youth_application: YouthApplication = self.get_object().lock_for_update()
         # Update unhandled youth applications' encrypted_handler_vtj_json so
         # handlers can accept/reject using it
-        if not youth_application.is_handled:
+        if (
+            youth_application.has_social_security_number
```

Comment:
Should it raise an issue if otherwise?

Reply:
No, because there's a side-effect of updating the encrypted_handler_vtj_json from VTJ whenever reading data from this endpoint (to the handler's UI usually). If the person has no social security number (no possibility to fetch any data from VTJ as it requires the social security number to fetch data) or the application has been handled (no need to fetch any data from VTJ as the handling has been done already) there's no possibility or no need to update the VTJ data.

Why is there such a side-effect in the first place? So that whatever VTJ data the handlers use when manually approving or rejecting youth applications gets saved, and also that the VTJ data is up to date when the handlers use them for approval/rejection.